Sources/Xcodeproj/create().swift

```diff
@@ -0,0 +1,115 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+
+//TODO our references should remain human readable, but they
+// aren't unique enough, eg. if a module is called ProxyFoo then the TargetProxy for Foo will conflict with the Target for ProxyFoo
+
+//TODO blue folder everything not part of the build
+
+//TODO split out tests and non-tests in Products group
+
+//TODO enable code coverage
+
+//TODO make frameworks instead of dylibs and command line toggle
+
+
+import PackageType
+import Utility
+import POSIX
+
+public func create(path path: String, package: Package, modules: [SwiftModule], products: [Product]) throws {
+
+    let dirname = try mkdir(path, "\(package.name).xcodeproj")
+
+    try fopen(dirname, "project.pbxproj", mode: .Write) { fp in
+        try print(package: package, modules: modules, products: products) { line in
+            try fputs(line, fp)
+            try fputs("\n", fp)
+        }
+    }
+
+    let schemedir = try mkdir(dirname, "xcshareddata/xcschemes")
+
+    try fopen(schemedir, "\(package.name).xcscheme", mode: .Write) { fp in
+
+        func write(line: String) throws {
+            try fputs(line, fp)
+            try fputs("\n", fp)
+        }
```

Comment:
If you move this `write` function to line 31, you'll be able to use it in both `print` of the project and of the scheme, instead of having two identical copies of the same code for adding a newline after each printed line.